STATUS: CLEANER DOES NOT FIX
MODIFIED: N/A
MODIFIED FILE: N/A
CLEANER: io.github.dddplus.runtime.registry.IntegrationTest.stepsExecTemplateOneStepTimeoutWillNotRollback
CLEANER FILE: /home/lichengpeng/test/final/automated_script/funkygao/cp-ddd-framework/dddplus-test/src/test/java/io/github/dddplus/runtime/registry/IntegrationTest.java
POLLUTER: io.github.dddplus.runtime.registry.BadOnPurposeTest.emptyCodeStep
POLLUTER FILE: /home/lichengpeng/test/final/automated_script/funkygao/cp-ddd-framework/dddplus-test/src/test/java/io/github/dddplus/runtime/registry/BadOnPurposeTest.java
ORIGINAL CLEANER SIZE: 11
NEW CLEANER SIZE: 11
ELAPSED TIME: 0.0
==========================
@@ -0,0 +0,27 @@
+ {
+     {
+     }
+     {
+         fooModel = new FooModel();
+         fooModel.setPartnerCode(FooPartner.CODE);
+         fooModel.setB2c(true);
+         // ExtensionInvocationHandler.extInvokeTimerExecutor的线程池缩小到10，方便并发测试
+         System.setProperty("invokeExtMaxPoolSize", "10");
+     }
+     fooModel.setB2c(false);
+     fooModel.setRedecide(false);
+     fooModel.setStepsRevised(false);
+     fooModel.setSleepExtTimeout(true);
+     List<String> steps = DDD.findAbility(DecideStepsAbility.class).decideSteps(fooModel, Steps.Submit.Activity);
+     // FooStep调用一个扩展点超时，会抛出 ExtTimeoutException，不确定状态：抛出到外面，框架层不做回滚
+     try {
+         submitStepsExec.execute(Steps.Submit.Activity, steps, fooModel);
+         fail();
+     } catch (ExtTimeoutException expected) {
+         assertEquals("timeout:100ms", expected.getMessage());
+     }
+     {
+     }
+     {
+     }
+ }
